context/user_plane_information_test.go

```diff
@@ -219,17 +221,92 @@ func TestGenerateDefaultPath(t *testing.T) {
 	}
 }
 
-func TestGetDefaultUPFTopoByDNN(t *testing.T) {
-}
+func TestUpdateSmfUserPlaneNode_NodeIDChange(t *testing.T) {
+	upi := &context.UserPlaneInformation{
+		UPNodes:              make(map[string]*context.UPNode),
+		UPFs:                 make(map[string]*context.UPNode),
+		AccessNetwork:        make(map[string]*context.UPNode),
+		UPFIPToName:          make(map[string]string),
+		UPFsID:               make(map[string]string),
+		UPFsIPtoID:           make(map[string]string),
+		DefaultUserPlanePath: make(map[string][]*context.UPNode),
+	}
 
-func TestInsertSmfUserPlaneNode(t *testing.T) {
+	nodeID := pfcpType.NodeID{
+		NodeIdType:  pfcpType.NodeIdTypeIpv4Address,
+		NodeIdValue: []byte(net.ParseIP("1.2.3.4").To4()),
+	}
 
-}
+	// Create an existing UPNode with a specific UPF instance
+	originalUPF := context.NewUPF(&nodeID, nil)
+	existingNode := &context.UPNode{
+		Type:   "UPF",
+		NodeID: nodeID,
+		Port:   1234,
+		UPF:    originalUPF,
+		Links: []*context.UPNode{
+			{
+				Type: context.UPNODE_AN,
```

Comment:
Given that there is more code in the `case UPNODE_UPF:`, would it be better to have this test for `UPNODE_UPF` instead of `UPNOD_AN`?

Reply:
We could, the reason I used the `AN` is that it was with such a link that I noticed the issue.